src/main/java/net/consensys/orion/http/handler/receive/ReceiveHandler.java

```diff
@@ -79,12 +83,18 @@ public void handle(final RoutingContext routingContext) {
         return;
       }
 
-      final byte[] decryptedPayload;
-      try {
-        decryptedPayload = enclave.decrypt(encryptedPayload.get(), recipient);
-      } catch (final EnclaveException e) {
-        log.info("unable to decrypt payload with key {}", key);
-        routingContext.fail(404, new OrionException(OrionErrorCode.ENCLAVE_KEY_CANNOT_DECRYPT_PAYLOAD, e));
+      byte[] decryptedPayload = null;
+      EnclaveException exception = null;
+      for (final Box.PublicKey recipient : recipients) {
+        try {
+          decryptedPayload = enclave.decrypt(encryptedPayload.get(), recipient);
+        } catch (final EnclaveException e) {
+          exception = e;
+        }
+      }
+      if (decryptedPayload == null) {
+        log.info("unable to decrypt payload");
+        routingContext.fail(404, new OrionException(OrionErrorCode.ENCLAVE_KEY_CANNOT_DECRYPT_PAYLOAD, exception));
         return;
       }
```

Comment:
This block looks like a function in itself to try decrypt the payload

Reply:
Done